modules/mono/csharp_script.cpp

```diff
@@ -3000,107 +2079,102 @@ void CSharpScript::_bind_methods() {
+	Dictionary rpc_functions_dict;
+	rpc_functions_dict.~Dictionary();
+	Dictionary signals_dict;
+	signals_dict.~Dictionary();
 
-	p_script->tool = p_script->script_class->has_attribute(CACHED_CLASS(ToolAttribute));
+	Ref<CSharpScript> base_script;
+	GDMonoCache::managed_callbacks.ScriptManagerBridge_UpdateScriptClassInfo(
+			p_script.ptr(), &tool, &methods_dict, &rpc_functions_dict, &signals_dict, &base_script);
 
-	if (!p_script->tool) {
-		GDMonoClass *nesting_class = p_script->script_class->get_nesting_class();
-		p_script->tool = nesting_class && nesting_class->has_attribute(CACHED_CLASS(ToolAttribute));
-	}
+	p_script->tool = tool;
 
-#ifdef TOOLS_ENABLED
-	if (!p_script->tool) {
-		p_script->tool = p_script->script_class->get_assembly() == GDMono::get_singleton()->get_tools_assembly();
-	}
-#endif
+	p_script->rpc_config.clear();
+	p_script->rpc_config = rpc_functions_dict;
 
-#ifdef DEBUG_ENABLED
-	// For debug builds, we must fetch from all native base methods as well.
-	// Native base methods must be fetched before the current class.
-	// Not needed if the script class itself is a native class.
+	// Methods
 
-	if (p_script->script_class != p_script->native) {
-		GDMonoClass *native_top = p_script->native;
-		while (native_top) {
-			native_top->fetch_methods_with_godot_api_checks(p_script->native);
+	p_script->methods.clear();
 
-			if (native_top == CACHED_CLASS(GodotObject)) {
-				break;
-			}
+	for (const Variant *s = methods_dict.next(nullptr); s != nullptr; s = methods_dict.next(s)) {
+		StringName name = *s;
+
+		MethodInfo mi;
+		mi.name = name;
+
+		Array params = methods_dict[*s];
 
-			native_top = native_top->get_parent_class();
+		for (int i = 0; i < params.size(); i++) {
+			Dictionary param = params[i];
+
+			Variant::Type param_type = (Variant::Type)(int)param["type"];
+			PropertyInfo arg_info = PropertyInfo(param_type, (String)param["name"]);
+			arg_info.usage = (uint32_t)param["usage"];
+			mi.arguments.push_back(arg_info);
 		}
+
+		p_script->methods[name] = mi;
 	}
-#endif
 
-	p_script->script_class->fetch_methods_with_godot_api_checks(p_script->native);
+	// Event signals
 
-	p_script->rpc_config.clear();
+	// Performance is not critical here as this will be replaced with source generators.
 
-	GDMonoClass *top = p_script->script_class;
-	while (top && top != p_script->native) {
-		// Fetch methods from base classes as well
-		top->fetch_methods_with_godot_api_checks(p_script->native);
+	p_script->event_signals.clear();
 
-		// Update RPC info
-		{
-			Vector<GDMonoMethod *> methods = top->get_all_methods();
-			for (int i = 0; i < methods.size(); i++) {
-				if (!methods[i]->is_static()) {
-					const Variant rpc_config = p_script->_member_get_rpc_config(methods[i]);
-					if (rpc_config.get_type() != Variant::NIL) {
-						p_script->rpc_config[methods[i]->get_name()] = rpc_config;
-					}
-				}
-			}
+	// Sigh... can't we just have capacity?
+	p_script->event_signals.resize(signals_dict.size());
```

Comment:
`LocalVector` seems to copy all the contents in the copy constructor. We currently return `event_signals` from a function and I don't trust it will be optimized. `Vector` does copy-on-write, so it's not a problem there.